### PR TITLE
Refactor streaming filtering logic and document it

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This is work on top of #26159 for @ClearlyClaire — I've attempted to simplify and document the logic applied during the filtering process, and just generally make it more readable as to what the heck is going on.

### CODE REVIEW

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:631]: @ClearlyClaire I ended up dropping the `queued_at` / Delay logging from here as it was only present in the `silly` log level and wasn't available for majority of events (as only a small fraction actually contain a queued_at timestamp, and it's unclear as to exactly what this is measuring as a metric.

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:657]: @ClearlyClaire can you double check my boolean logic here? I think I got it right for what we want:

> If the stream doesn't need filtering, or the event isn't a `update` or `status.update` event, then transmit it directly.

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:675]: @ClearlyClaire now that streaming is always behind authentication, we should be able to remove this check, as we should always have an `req.accountId` value (if we don't we should've failed the connection attempt)

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:719]:
```suggestion
          // as such, we don't need to construct or apply the filters in streaming:
```

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:634]: I'm not sure what the advantage of doing that here is, since we're already having the `event === 'update' || event === 'status.update'` logic in the place we are computing `filter_results` and passing the payload…

File: streaming/index.js
Comment (ClearlyClaire) [reply streaming/index.js:657]: That looks fine.

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:805]: Like, this would just be `transmit({ ...payload, filtered: filter_results })`.

File: streaming/index.js
Comment (ThisIsMissEm) [reply streaming/index.js:634]: You'll still need the event name for logging, as this function moved up in scope, but yeah, I guess now that the code is more documented, doing the `{ ...payload, filtered: filter_results }` at transmit call-site is better. Will update

File: streaming/index.js
Comment (ThisIsMissEm) [reply streaming/index.js:631]: Further, the `queued_at` value is removed entirely by https://github.com/mastodon/mastodon/pull/26173

**Comment by ThisIsMissEm** [comment]
@ClearlyClaire okay! Sorry for the delay, have updated this PR, and it should be good to go!

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:646]: These were just moved down to near the `pgPool.connect` below

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:773]: This can just be `if (req.cachedFilters)` because on line 713, we return early if the `payload` has already been filtered by the Ruby on Rails side (meaning `payload.filtered` is present)

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:642]:
```suggestion
      // Ruby on Rails side when producing the event for streaming.
```

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:712]:
```suggestion
          // that filters were applied on the Ruby on Rails side, as such, we
          // we don't need to construct or apply the filters in streaming:
```

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:775]:
```suggestion
            // TODO: Calculate searchableContent in Rails:
```

**Review by ClearlyClaire — COMMENTED** [review]
Thank you! It looks ok except for the comments about `mastodon-web` not being factual (most, if not all, of the Ruby on Rails side filtering is done in `mastodon-sidekiq`, and ultimately, I don't think making that distinction is very useful as the code base is shared between those two components).

Also, the more I think about it, the more I think this PR should be a standalone PR targeting `main` once #26159 is merged separately (though it can be merged immediately after #26159)

**Review by renchap — APPROVED** [review]
Looks good to me, please apply Claire's changes

**Pull request closed without merging**
